harness/assert.js

```diff
@@ -79,3 +79,12 @@ assert.throws = function (expectedErrorConstructor, func, message) {
   message += 'Expected a ' + expectedErrorConstructor.name + ' to be thrown but no exception was thrown at all';
   $ERROR(message);
 };
+
+assert.throws.early = function(err, code) {
+  var wrappedCode = `function wrapperFn() { ${code} }`;
+  var ieval = eval;
+
+  assert.throws(err, () => { Function(wrappedCode); }, `Function: ${code}`);
+  assert.throws(err, () => { eval(wrappedCode); }, `eval: ${code}`);
+  assert.throws(err, () => { ieval(wrappedCode); }, `indirect eval: ${code}`);
+};
```

Comment:
sure thing. I didn't want to write docs and tests while I was only requesting feedback. This is a prototype.

If we are moving this forward, I'll provide docs and tests before merging.